src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -360,7 +359,13 @@ impl<'a, 'tcx> PatternContext<'a, 'tcx> {
                     (PatternKind::Constant { value: lo },
                      PatternKind::Constant { value: hi }) => {
                         use std::cmp::Ordering;
-                        match (end, compare_const_vals(self.tcx, lo, hi, ty).unwrap()) {
+                        let cmp = compare_const_vals(
+                            self.tcx,
+                            lo,
+                            hi,
+                            self.param_env.and(ty),
+                        ).unwrap();
```

Comment:
Note that this `unwrap` can fail - we should be matching on `Option<Ordering>` below and explicitly list all the possible cases - currently the `_` success at the end of the `match` would catch `None` which would be bad.